library/alloc/src/boxed.rs

```diff
@@ -1242,8 +1240,8 @@ unsafe impl<#[may_dangle] T: ?Sized, A: Allocator> Drop for Box<T, A> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: Default> Default for Box<T> {
     /// Creates a `Box<T>`, with the `Default` value for T.
+    #[inline]
     fn default() -> Self {
-        #[rustc_box]
```

Comment:
The "`T` is already a function argument" justification doesn't apply here.

But what _does_ matter is that `T` is the return value for the `T::default()` call. So this use is slightly more exciting than the other cases.

Initially, I was concerned that this might require `#[rustc_box]`. But in a totally unoptimized build, you still cannot `Box::default` a `T` which is more than half your stack size: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e4322542645a0afe6c2644511b838e5c

One possible complaint is that this halves the size of the `T` that can be defaulted, because now we have to pass it through two function calls. That is true. But MIR optimizations can _also_ halve the possible size of `T`, if `T::default()` is inlined into `Box::default`. We don't do MIR inlining currently at `-Copt-level=0`, but my point is that anyone relying on the size of a `T` they can default with this function is already lined up for trouble outside of this PR.